parsers/CEB.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+"""Quarter-hourly data parser for Sri Lanka
+Fetches data for the previous day in 15-minute increments
+Data is from the backend for the load curve graph on https://cebcare.ceb.lk/gensum/details
+"""
+
+import json
+from datetime import datetime, timezone
+from logging import Logger, getLogger
+from zoneinfo import ZoneInfo
+
+# The request library is used to fetch content through HTTP
+from requests import Session
+
+from electricitymap.contrib.lib.models.event_lists import ProductionBreakdownList
+from electricitymap.contrib.lib.models.events import ProductionMix
+from electricitymap.contrib.lib.types import ZoneKey
+from parsers.lib.exceptions import ParserException
+
+TIMEZONE = ZoneInfo("Asia/Colombo")
+GENERATION_BREAKDOWN_URL = "https://cebcare.ceb.lk/GenSum/GetLoadCurveData"
+SOURCE_NAME = "ceb.lk"
+
+PRODUCTION_MAPPING = {
+    "SPP Biomass": "biomass",
+    "Coal": "coal",
+    "Thermal-Oil": "oil",
+    "SPP Minihydro": "hydro",
+    "Major Hydro": "hydro",
+    "Solar": "solar",
+    "Wind": "wind",
+}
+
+
+def fetch_production(
+    zone_key: ZoneKey = ZoneKey("LK"),
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+):
+    """Requests the previous day's production mix (in MW) for Sri Lanka, per quarter-hour"""
+    if target_datetime is None:
+        target_datetime = datetime.now(tz=timezone.utc)
+
+    ses = session or Session()
```

Comment:
```suggestion
    if session is None:
        session = Session()
```
I would recommend avoiding renaming to shorter variable names. It often can result in confusions. The more you can read your code without thinking about what this variable is, the better!